futures-util/src/async_await/select.rs

```diff
@@ -1,5 +1,30 @@
 //! The `select` macro.
 
+/// Polls multiple futures simultaneously, executing the branch for the future 
+/// that finishes first.
+/// 
+/// `select!` can select over futures with different output types, but each 
+/// branch has to resolve to the same type. Inside each branch, the respective 
```

Comment:
> each branch has to resolve to the same type.

- The word resolve is a good choice. "resolve" is usually only used for future. "return" would be a better choice
- Is this true if the output is not used?

Reply:
if the output is not used, the return type of the branch is `()`, isn't it?